addons/binding/org.openhab.binding.homie/src/main/java/org/openhab/binding/homie/internal/MqttConnection.java

```diff
@@ -0,0 +1,142 @@
+            client.connect(opts);
+            logger.debug("Homie MQTT Connection connected");
+        } catch (MqttException e) {
+            logger.error("MQTT Connect failed", e);
+        }
+    }
+
+    public void subscribe(Thing thing, IMqttMessageListener messageListener) throws MqttException {
+        String topic = String.format("%s/%s/#", basetopic, thing.getUID().getId());
+        resubscribe(topic, messageListener);
+
+    }
+
+    public void listenForDeviceIds(IMqttMessageListener messageListener) {
+        logger.debug("Listening for devices on topic {}", listenDeviceTopic);
+        try {
+            resubscribe(listenDeviceTopic, messageListener);
+        } catch (MqttException e) {
+            logger.error("Failed to subscribe to topic {}", listenDeviceTopic, e);
+        }
+    }
+
+    public void unsubscribeListenForDeviceIds() {
+        try {
+            client.unsubscribe(listenDeviceTopic);
+        } catch (MqttException e) {
+            logger.error("Failed to unsubscribe from topic {}", listenDeviceTopic, e);
+        }
+
+    }
+
+    /**
+     * Subscribe to a channel of a device
+     *
+     * @param channelUID
+     * @param handler
+     */
+    public void subscribeChannel(Channel channel, HomieDeviceHandler handler) {
+
+        String topic = String.format("%s/%s/%s", basetopic, handler.getThing().getUID().getId(),
+                channel.getProperties().get(CHANNELPROPERTY_TOPICSUFFIX));
+        logger.debug("(Re-)Subscribing to topic '{}' to listen for events of channel '{}'", topic, channel.getUID());
+        try {
+            resubscribe(topic, handler);
+            logger.debug("Subscribed to topic {}", topic);
+        } catch (MqttException e) {
+            logger.error("Error (re)subscribing to channel. topic is {}", topic, e);
+        }
+
+    }
+
+    private void resubscribe(String topic, IMqttMessageListener listener) throws MqttException {
+        client.unsubscribe(topic);
+        client.subscribe(topic, SUBSCRIBE_QOS, listener);
+    }
+
+    public void listenForNode(String deviceId, String nodeId, IMqttMessageListener listener) throws MqttException {
+        String topic = String.format("%s/%s/%s/", basetopic, deviceId, nodeId);
+        // Subscribe to type and proplist separately, as the notifications have to arrive first at the listener
+        resubscribe(topic + HomieConventions.HOMIE_NODE_TYPE_ANNOUNCEMENT_TOPIC_SUFFIX, listener);
+        resubscribe(topic + HomieConventions.HOMIE_NODE_PROPERTYLIST_ANNOUNCEMENT_TOPIC_SUFFIX, listener);
+        resubscribe(topic + "#", listener);
+    }
+
+    public void send(String deviceId, String nodeId, String property, Command command)
+            throws MqttPersistenceException, MqttException {
+        String topic = String.format("%s/%s/%s/%s/set", basetopic, deviceId, nodeId, property);
+        client.publish(topic, command.toString().getBytes(), PUBLISH_QOS, true);
```

Comment:
Better always provide an encoding when calling getBytes otherwise behaviour becomes platform dependent, can you check all?